Add forward compatible Test interface

b1rdex opened the pull request:
5.7 is still alive and it misses much of forward compatibility.

b1rdex committed:
Create Test.php

sebastianbergmann commented:
Thank you for your contribution. I appreciate the time you invested in preparing this pull request. However, I have decided not to merge it.

The forward compatibility layer will not be extended. It is intended to only cover the most common use case, which is basic usage of `PHPUnit\Framework\TestCase`.

b1rdex commented:
Sad. Let me argue why I added that.

I added this because forward compatibility layer has `TestListener` that requires `PHPUnit_Framework_Test` in it's signature. The only way to solve this — add class alias with such name (class extension will not pass signature checks).

Writing `Listener`s that are compatible with phpunit 5+6 is impossible without that aliasing and this hack is added to client code, which is not correct.

sebastianbergmann commented:
Thank you for clarifying. I'll extend the forward compatibility layer with `AssertionFailedError`, `Test`, and `TestSuite` to ease enable the development of test listeners that are compatible with PHPUnit 5 and 6.
